docs/src/extend/custom-rules.md

```diff
@@ -787,6 +789,47 @@ module.exports = {
 
 To learn more about JSON Schema, we recommend looking at some examples on the [JSON Schema website](https://json-schema.org/learn/), or reading the free [Understanding JSON Schema](https://json-schema.org/understanding-json-schema/) ebook.
 
+#### Option Defaults
+
+Rules may specify a `meta.defaultOptions` array of default values for any options.
+When the rule is enabled in a user configuration, ESLint will recursively merge any user-provided option elements on top of the default elements.
+
+For example, given the following defaults:
+
+```js
+module.exports = {
+    meta: {
+        defaultOptions: [{
+            alias: "basic",
+        }],
+        schema: {
+            type: "object",
+            properties: {
+                alias: {
+                    type: "string"
+                }
+            },
+            additionalProperties: false
+        }
+    },
+    create(context) {
+        const [{ alias }] = context.options;
+
+        return { /* ... */ };
+    }
+}
+```
+
+The rule would have a runtime `alias` value of `"basic"` unless the user configuration specifies a different value, such as with `["error", { alias: "complex" }]`.
+
+Each element of the options array is merged according to the following rules:
+
+* User-provided `undefined` uses any default option
```

Comment:
The wording of this bullet is a bit confusing to me. Do you mean a hardcoded `undefined` or an undefined as in it wasn't specified? And by "any default option", I think you mean that the specified default in `defaultOptions` is used?

Reply:
Heh, yeah, this confused me re-reading it.